lib/editor/result.js

```diff
@@ -0,0 +1,281 @@
+'use babel'
+import {CompositeDisposable, Emitter} from 'atom'
+import ResultView from './result-view'
+import fastdom from 'fastdom'
+
+let layers = {}
+
+export default class Result {
+  constructor (editor, lineRange, opts) {
+    this.emitter = new Emitter()
+    this.disposables = new CompositeDisposable()
+
+    this.editor = editor
```

Comment:
I've definitely noticed that things slow down after a while in Juno. But it doesn't matter for the sake of this PR.